classes/log/store.php

```diff
@@ -63,6 +64,56 @@ protected function is_event_ignored(event_base $event) {
         return $isdisabledevent;
     }
 
+    /**
+     * Get the eventid from logstore_standard_log.
+     * Return the persistent logstore_standard_log id from the database.
+     * Return value 0 means not found.
+     * 
+     * @param event_base $event
+     * @return int
+     *
+     */
+    protected function get_event_id($event) {
+        global $DB;
+
+        $sqlparams = [
+            'eventname' => $event->eventname, 
+            'component' => $event->component, 
+            'action' => $event->action, 
+            'target' => $event->target, 
+            'objecttable' => $event->objecttable, 
+            'objectid' => $event->objectid, 
+            'timecreated' => $event->timecreated, 
+            'userid' => $event->userid, 
+            'anonymous' => $event->anonymous
+        ];
+
+        $sql = "SELECT MAX(ID) AS id
+                  FROM {logstore_standard_log}
+                 WHERE eventname = :eventname
+                   AND component = :component
+                   AND action = :action
+                   AND target = :target
+                   AND (objecttable = :objecttable OR objecttable IS NULL)
+                   AND (objectid = :objectid OR objectid IS NULL)
+                   AND timecreated = :timecreated
+                   AND userid = :userid
+                   AND anonymous = :anonymous";
+
+        $rows = $DB->get_records_sql($sql, $sqlparams);
```

Comment:
`MAX(ID)` should be `MAX(id)`?

If we only want 1 record then surely it would make more sense to use `get_record_sql'? Then you could remove the foreach below and replace with:
```
if (!empty($row)) {
   $rowid = $row->id;
}
```

Reply:
MAX(ID) still works but I'll change it to MAX(id). I was wondering if it would be possible that multiple rows could be returned by double logging in the same millisecond, so that's why I return the MAX(id). Originally it returned multiple rows so I stuck with get_records_sql().

Reply:
I have changed the code to use get_record_sql(); it saves a few lines of code.
Is it standard to use:
```
        if (empty($row) || empty($row->id)) {
            return 0;
        }
        return $row->id;
```
I tested it and it seems OK. No need to iterate the rows or use array_keys.